osltweaks.php

```diff
@@ -16,6 +16,35 @@ function osltweaks_civicrm_buildForm($formName, &$form) {
       // include some javascript to finagle the wording in that section.
       CRM_Core_Resources::singleton()->addScriptFile('org.osltoday.osltweaks', 'js/CRM_Contribute_Form_Contribution_Main-isShowCMS.js');
     }
+
+    // Array of Contribution page id for the Country and State rewrite
+    $settings = CRM_Core_BAO_Setting::getItem(NULL, 'com.joineryhq.osltweaks');
+
+    foreach ($settings['page_ids'] as $key => $id) {
+      // Check if contribution page id is match on the $settings and make changes
+      if ($id == $_GET['id']) {
+        if ( $form->elementExists( 'country-1' ) ) {
+          // Remove other Country and show only US
+          $elCountry = & $form->getElement('country-1');
+          $elCountryOptions = & $elCountry->_options;
+          foreach ($elCountryOptions as $key => $option) {
+            if ( $option['attr']['value'] != 1228 ) {
+              unset($elCountryOptions[$key]);
+            }
+          }
+
+          // Rewrite State/Country label to State
+          $elState = & $form->getElement('state_province-1');
+          $elState->_label = 'State';
+
+          // Set Country to US and delete value of State if its US Contribution and User Country is not
+          if ($elCountry->_values[0] != 1228) {
+            $elCountry->_values[0] = 1228;
```

Comment:
Can you leave the country field empty (`unset()`) in this case, as you're doing with State?

Reply:
`unset()` on `$elCountry` will have a bug on the state field. No changes made for this comment.